File: SPA/FrontEnd/src/app/Models/PuntoTuristicoCategoria/punto-turistico-categoria.ts
import { Categoria } from '../Categoria/categoria';
import { PuntoTuristico } from '../Punto/punto';
export class PuntoTuristicoCategoria {
    puntoTuristicoId: number;
    puntoTuristico: PuntoTuristico;
    categoriaId: number;
    categoria: Categoria;
    constructor(puntoId: number, categoriaId: number, punto: PuntoTuristico, categoria: Categoria) {
        this.categoria = categoria;
        this.categoriaId = categoriaId;
        this.puntoTuristicoId = puntoId;
        this.puntoTuristico = punto;
    }
}

File: Domain/PuntoTuristicoCategoria.cs
using BuisnessLogic.Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain
{
    public class PuntoTuristicoCategoria
    {
        public int PuntoTuristicoId { get; set; }
        public virtual PuntoTuristico PuntoTuristico { get; set; }
        public int CategoriaId { get; set; }
        public virtual Categoria Categoria { get; set; }
        public override bool Equals(object obj)
        {
            try
            {
                PuntoTuristicoCategoria comparar = (PuntoTuristicoCategoria)obj;
                return this.CategoriaId == comparar.CategoriaId && this.PuntoTuristicoId == comparar.PuntoTuristicoId;
                //no pueden haber 2 puntos con misma id. Analogo para categorias
            }
            catch (Exception)
            {
                return false;
            }
        }
    }
}
